hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/IsAssociatedSystemContractTest.java

```diff
@@ -0,0 +1,318 @@
+                    .andAssert(query -> query.has(ContractFnResultAsserts.resultWith()
+                            .resultThruAbi(
+                                    getABIFor(FUNCTION, "isAssociated", "HRCContract"),
+                                    isLiteralResult(new Object[] {false})))));
+        }
+    }
+
+    @Nested
+    @DisplayName("returns false")
+    class ReturnsFalse {
+        static final AtomicReference<String> fungibleTokenNum = new AtomicReference<>();
+        static final AtomicReference<String> nonFungibleTokenNum = new AtomicReference<>();
+        private static final String hrcContract = "HRCContract";
+
+        @BeforeAll
+        static void beforeAll(@NonNull final TestLifecycle testLifecycle) {
+            testLifecycle.doAdhoc(
+                    cryptoCreate("account").balance(100 * ONE_HUNDRED_HBARS),
+                    uploadInitCode(hrcContract),
+                    contractCreate(hrcContract),
+                    newKeyNamed("multikey"),
+                    tokenCreate("fungibleToken").treasury("account").exposingCreatedIdTo(fungibleTokenNum::set),
+                    tokenCreate("nonFungibleToken")
+                            .tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+                            .treasury("account")
+                            .initialSupply(0)
+                            .supplyKey("multikey")
+                            .exposingCreatedIdTo(nonFungibleTokenNum::set));
+        }
+
+        @HapiTest
+        @DisplayName("when token is not associated with a contract")
+        public Stream<DynamicTest> checkTokenIsNotAssociatedWithContract() {
+            return hapiTest(
+                    sourcing(() -> contractCall(
+                                    hrcContract,
+                                    "isAssociated",
+                                    asHeadlongAddress(asHexedSolidityAddress(asToken(fungibleTokenNum.get()))))
+                            .gas(1_000_000)
+                            .via("test")),
+                    childRecordsCheck(
+                            "test",
+                            SUCCESS,
+                            TransactionRecordAsserts.recordWith()
+                                    .status(SUCCESS)
+                                    .contractCallResult(resultWith()
+                                            .resultThruAbi(
+                                                    getABIFor(
+                                                            Utils.FunctionType.FUNCTION, "isAssociated", "HRCContract"),
+                                                    isLiteralResult(new Object[] {false})))));
+        }
+
+        @HapiTest
+        @DisplayName("when nft is not associated with a contract")
+        public Stream<DynamicTest> checkNftIsNotAssociatedWithContract() {
+            return hapiTest(
+                    sourcing(() -> contractCall(
```

Comment:
Many of these tests have basically the same structure.  Is it possible to create a static helper which is largely the same and just call that with a few parameters?